### ISSUE TITLE
Breaking change found related to streams from slbeta2 -> slbeta3

### ISSUE THREAD

**Issue by LakshanSS** [ballerina-platform/ballerina-lang#31994]
**Description:**
Most of the connector daily builds are failing in `slbeta3 SNAPSHOT`
There is a compilation error related to the usage of streams.

These connectors were working fine in `slbeta2`

**Sample Error logs:**
```
ERROR [endpoints.bal:(56:43,56:57)] incompatible types: expected '(record {| ballerinax/googleapis.calendar:0.1.6-SNAPSHOT:Calendar value; |}|error)', found '(record {| ballerinax/googleapis.calendar:0.1.6-SNAPSHOT:Calendar value; |}|error)?'
ERROR [endpoints.bal:(178:40,178:51)] incompatible types: expected '(record {| ballerinax/googleapis.calendar:0.1.6-SNAPSHOT:Event value; |}|error)', found '(record {| ballerinax/googleapis.calendar:0.1.6-SNAPSHOT:Event value; |}|error)?'
error: compilation contains errors
```

**Sample link to failing daily build workflow:**
https://github.com/ballerina-platform/module-ballerinax-googleapis.calendar/runs/3307524894?check_suite_focus=true

**Reason for this breaking change:**
The fixes done to solve https://github.com/ballerina-platform/ballerina-lang/issues/31464 might have created this breaking change in `slbeta3`

**Affected Versions:**
Ballerina slbeta3-SNAPSHOT

**OS, DB, other environment details and versions:**
JDK 11
Ubuntu

**Comment by LakshanSS** [comment]
@anupama-pathirage Should we do the changes in connector side? or can we get it fixed from ballerina side?

It seems like if we have do the changes in connector side, we have to change the return type of the remote functions.

Example:
`returns stream<Calendar,error>|error` to `returns stream<Calendar,error?>|error`

If we do a method signature change in connector, this can break existing choreo-samples and custom integrations built by users.

Or else, is there any other way to fix this without changing the connector remote function signutures?

Please share your thoughts.

**Comment by anupama-pathirage** [comment]
@hasithaa / @gimantha Please check on this and can you share more thoughts on this please. Did we introduce any breaking change?

**Comment by pcnfernando** [comment]
Yes, we had to introduce https://github.com/ballerina-platform/ballerina-lang/pull/31674 breaking change. 
With beta2 there was a bug where we had considered, `stream<T, E>` as equivalent to `stream<T, E?>`. We had handled the nil completion type internally. Due to this, we encountered crashes at runtime eg: https://github.com/ballerina-platform/ballerina-lang/issues/31671

Even though we warned this change is coming before beta2, seems we have missed this change in some modules as the compiler restriction was not there.
https://github.com/ballerina-platform/ballerina-lang/pull/31674
